src/app/employees/employee-home/employee-home.component.ts

```diff
@@ -39,25 +38,29 @@ import { Employee } from '../employee';
 export class EmployeeHomeComponent implements OnInit {
   public employees: Employee[];
 
-  constructor(
-    public loginService: LoginService,
-    private dataService: DataService) {
-  }
+  constructor(private dataService: DataService) { }
+
   message: boolean;
   showMobileMenu = false;
   showTopMenu = false;
   showPracticeMenu = false;
   showClientMenu = false;
 
   ngOnInit() {
-    this.employees = this.dataService
-      .getEmployees()
-      .slice()
-      .filter(x => x.dayssincehire <= 180)
-      .sort((a, b) => {
-        const x = new Date(a.anniversary);
-        const y = new Date(b.anniversary);
-        return y.getTime() - x.getTime();
+    this.dataService.fetchEmployees()
+      .subscribe((employees: Employee[]) => {
+        this.dataService.processEmployees(employees);
```

Comment:
does the app break if I enter from a different point because the processEmployees call is here?  by that, I mean.  what if i navigate diretly to /employee/1133 for example?

Reply:
> does the app break if I enter from a different point because the processEmployees call is here? by that, I mean. what if i navigate diretly to /employee/1133 for example?

I think it's already broken. When I do the same thing in production, it redirects to the home page and not the detail page. It must be in the way the routes are set up.